prime-dotnet-webapi/Controllers/BannersController.cs

```diff
@@ -24,111 +25,163 @@ public BannersController(IBannerService bannerService)
-        [HttpPut("site-landing", Name = nameof(SetSiteLandingBanner))]
+        [HttpPost("enrolment-landing", Name = nameof(CreateEnrolleeLandingBanner))]
         [Authorize(Roles = Roles.PrimeSuperAdmin)]
         [ProducesResponseType(StatusCodes.Status401Unauthorized)]
         [ProducesResponseType(StatusCodes.Status403Forbidden)]
         [ProducesResponseType(typeof(ApiResultResponse<BannerViewModel>), StatusCodes.Status200OK)]
-        public async Task<ActionResult> SetSiteLandingBanner(BannerViewModel viewModel)
+        public async Task<ActionResult> CreateEnrolleeLandingBanner(BannerViewModel viewModel)
         {
-            var updatedBanner = await _bannerService.SetBannerAsync(BannerLocationCode.SiteRegistrationLandingPage, viewModel);
-            return Ok(updatedBanner);
+            var banner = await _bannerService.CreateBannerAsync(BannerLocationCode.EnrolmentLandingPage, viewModel);
+            return Ok(banner);
         }
 
-        // GET: api/banners/enrolment-landing
+        // POST: api/banners/site-landing
         /// <summary>
-        /// Get the Enrollee landing banner
+        /// Set the Site landing banner
         /// </summary>
-        [HttpGet("enrolment-landing", Name = nameof(GetEnrolleeLandingBanner))]
+        /// <param name="viewModel"></param>
+        [HttpPost("site-landing", Name = nameof(CreateSiteLandingBanner))]
         [Authorize(Roles = Roles.PrimeSuperAdmin)]
         [ProducesResponseType(StatusCodes.Status401Unauthorized)]
         [ProducesResponseType(StatusCodes.Status403Forbidden)]
         [ProducesResponseType(typeof(ApiResultResponse<BannerViewModel>), StatusCodes.Status200OK)]
-        public async Task<ActionResult> GetEnrolleeLandingBanner()
+        public async Task<ActionResult> CreateSiteLandingBanner(BannerViewModel viewModel)
         {
-            var banner = await _bannerService.GetBannerAsync(BannerLocationCode.EnrolmentLandingPage);
+            var banner = await _bannerService.CreateBannerAsync(BannerLocationCode.SiteRegistrationLandingPage, viewModel);
             return Ok(banner);
         }
 
-        // GET: api/banners/site-landing
+        // PUT: api/banners/1
         /// <summary>
-        /// Get the Site landing banner
+        /// Update banner
         /// </summary>
-        [HttpGet("site-landing", Name = nameof(GetSiteLandingBanner))]
+        /// <param name="bannerId"></param>
+        /// <param name="viewModel"></param>
+        [HttpPut("{bannerId}", Name = nameof(UpdateBanner))]
         [Authorize(Roles = Roles.PrimeSuperAdmin)]
         [ProducesResponseType(StatusCodes.Status401Unauthorized)]
         [ProducesResponseType(StatusCodes.Status403Forbidden)]
         [ProducesResponseType(typeof(ApiResultResponse<BannerViewModel>), StatusCodes.Status200OK)]
-        public async Task<ActionResult> GetSiteLandingBanner()
+        public async Task<ActionResult> UpdateBanner(int bannerId, BannerViewModel viewModel)
```

Comment:
Since Id is in the View Model now, pragmatically we should be performing a little validation here to make sure the Id matches the Id in the View Model.